x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/settings.tsx

```diff
@@ -53,59 +41,166 @@ export function Settings({
-      </EuiContextMenuItem>
-    ));
+  const sideNavItems = [
+    {
+      name: '',
+      id: '0',
+      items: getOperationTypes(filteredColumns, selectedColumn).map(
+        ({ operationType, compatibleWithCurrentField }) => ({
+          name: operationPanels[operationType].displayName,
+          id: operationType as string,
+          className: classNames('lnsConfigPanel__operation', {
+            'lnsConfigPanel__operation--selected': Boolean(
+              incompatibleSelectedOperationType === operationType ||
+                (!incompatibleSelectedOperationType &&
+                  selectedColumn &&
+                  selectedColumn.operationType === operationType)
+            ),
+            'lnsConfigPanel__operation--incompatible': !compatibleWithCurrentField,
+          }),
+          'data-test-subj': `lns-indexPatternDimension-${operationType}`,
+          onClick() {
+            if (!selectedColumn || !compatibleWithCurrentField) {
+              setInvalidOperationType(operationType);
+              return;
+            }
+            if (incompatibleSelectedOperationType) {
+              setInvalidOperationType(null);
+            }
+            if (selectedColumn.operationType === operationType) {
+              return;
+            }
+            const newColumn: IndexPatternColumn = filteredColumns.find(
+              col =>
+                col.operationType === operationType &&
+                (!hasField(col) ||
+                  !hasField(selectedColumn) ||
+                  col.sourceField === selectedColumn.sourceField)
+            )!;
+            setState(changeColumn(state, columnId, newColumn));
+          },
+        })
+      ),
+    },
+  ];
 
-  return selectedColumn && (operationMenuItems.length > 1 || contextOptions.length > 0) ? (
-    <EuiFlexItem grow={null}>
-      <EuiPopover
-        id={columnId}
-        panelClassName="lnsIndexPattern__dimensionPopover"
-        isOpen={isSettingsOpen}
-        closePopover={() => {
-          setSettingsOpen(false);
-        }}
-        ownFocus
-        anchorPosition="leftCenter"
-        panelPaddingSize="none"
-        button={
-          <EuiFlexItem data-test-subj="indexPattern-dimension" grow={true}>
-            <EuiButtonIcon
-              data-test-subj="indexPattern-dimensionPopover-button"
-              onClick={() => {
-                setSettingsOpen(!isSettingsOpen);
-              }}
-              iconType="gear"
-              aria-label={i18n.translate('xpack.lens.indexPattern.settingsLabel', {
-                defaultMessage: 'Settings',
+  return (
+    <EuiPopover
+      id={columnId}
+      className="lnsConfigPanel__summaryPopover"
+      anchorClassName="lnsConfigPanel__summaryPopoverAnchor"
+      button={
+        <EuiLink
+          className="lnsConfigPanel__summaryLink"
+          onClick={() => {
+            setSettingsOpen(true);
+          }}
+          data-test-subj="indexPattern-configure-dimension"
+        >
```

Comment:
In the mocks, there is also an icon indicating the data type of the column. Your choice on the timing of adding that.

Reply:
I will do this together with the data panel styling PR, as we discussed offline there might be re-usable pieces